### PR TITLE
[CodeCompletion] Associate callee declaration with call patterns

### PR DESCRIPTION
**Pull request by rintaro**
In `addFunctionCallPattern()`, we should associate the declaration if possible.

rdar://problem/49158044

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test OS X platform

**Comment by rintaro** [comment]
@swift-ci Please smoke test OS X platform

**Comment by rintaro** [comment]
https://ci.swift.org/job/swift-PR-osx-smoke-test/13926/console
I believe lldb failures are not related to this PR.
@shahmishal Have you seen this before?

**Comment by rintaro** [comment]
@swift-ci Please smoke test macOS

File: lib/IDE/ExprContextAnalysis.cpp [hunk 1]
```diff
@@ -241,7 +242,7 @@ namespace {
 class ExprParentFinder : public ASTWalker {
   friend class ExprContextAnalyzer;
   Expr *ChildExpr;
-  llvm::function_ref<bool(ParentTy, ParentTy)> Predicate;
+  std::function<bool(ParentTy, ParentTy)> Predicate;
```
Comment (benlangmuir) [inline lib/IDE/ExprContextAnalysis.cpp]: Why is this change required?

**Review by benlangmuir — APPROVED** [review]

File: lib/IDE/ExprContextAnalysis.cpp (hunk 1 above)
Comment (rintaro) [reply lib/IDE/ExprContextAnalysis.cpp]: Because it crashes without this change. `llvm::function_ref` doesn't really support usage like this: http://llvm.org/doxygen/STLExtras_8h_source.html#l00110
> This is intended for use as the type of a function parameter that is not used after the function in question returns.

I think it worked because it didn't capture any outer values. Now, it references outer values `SM` and `ParsedExpr`.

**Comment by rintaro** [comment]
@swift-ci Please smoke test
